tests/lib/rules/no-implicit-coercion.js

```diff
@@ -46,6 +51,15 @@ ruleTester.run("no-implicit-coercion", rule, {
         {code: "+foo.bar", errors: [{message: "use `Number(foo.bar)` instead.", type: "UnaryExpression"}]},
         {code: "1*foo", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
         {code: "foo*1", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*foo*1", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*1*foo", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*bar*1*foo", errors: [
+            {message: "use `Number(bar)` instead.", type: "BinaryExpression"},
+            {message: "use `Number(foo)` instead.", type: "BinaryExpression"}
+        ]},
+        {code: "1234*1*foo*bar", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
```

Comment:
It is not about being unnecessary. It is about casting the variable. Anywhere `1 * something` is a potential casting.